File: tools/autoLimit/programs/program2.py
#!/usr/bin/env python2.7
# -*- coding: utf-8 -*-

# This program is not meant to be time nor memory efficient.
# It is a reference program doing computations and using memory.

from __future__ import print_function

MAX_NUMBER = 20000

primes = []
for i in range(2, MAX_NUMBER):
    for j in range(2, i):
        if i % j == 0:
            primes.append(False)
            break
    else:
        primes.append(True)
        print("%d " % i, end='')
print("are primes.")

File: tools/autoLimit/programs/program2.ml
(* This program is not meant to be time nor memory efficient.
It is a reference program doing computations and using memory. *)

let max_number = 60000;;

let rec is_prime_rec n d =
    if d >= n
    then true
    else ((n mod d <> 0) && (is_prime_rec n (d+1)))
;;

let rec make_prime_list_rec n =
    if n > max_number
    then []
    else
      let ip = is_prime_rec n 2 in
      begin
        if ip
        then begin
          print_int n;
          print_char ' ';
          flush_all ();
        end;
        (is_prime_rec n 2)::(make_prime_list_rec (n+1))
      end
;;
make_prime_list_rec 2;;
